pkg/controller/job/job_controller.go

```diff
@@ -1710,28 +1713,28 @@ func matchPodFailurePolicy(job *batch.Job, failedPod *v1.Pod) (*string, bool) {
 				if containerStatus := matchOnExitCodes(&failedPod.Status, podFailurePolicyRule.OnExitCodes); containerStatus != nil {
 					switch podFailurePolicyRule.Action {
 					case batch.PodFailurePolicyActionIgnore:
-						return nil, true
+						return nil, false
 					case batch.PodFailurePolicyActionTerminate:
 						msg := fmt.Sprintf("Container %s for pod %s/%s failed with exit code %v matching %v rule at index %d",
 							containerStatus.Name, failedPod.Namespace, failedPod.Name, containerStatus.State.Terminated.ExitCode, podFailurePolicyRule.Action, index)
-						return &msg, false
+						return &msg, true
 					}
 				}
 			} else if podFailurePolicyRule.OnPodConditions != nil {
 				if podCondition := matchOnPodConditions(&failedPod.Status, podFailurePolicyRule.OnPodConditions); podCondition != nil {
 					switch podFailurePolicyRule.Action {
 					case batch.PodFailurePolicyActionIgnore:
-						return nil, true
+						return nil, false
 					case batch.PodFailurePolicyActionTerminate:
 						msg := fmt.Sprintf("Pod %s/%s has condition %v matching %v rule at index %d",
 							failedPod.Namespace, failedPod.Name, podCondition.Type, podFailurePolicyRule.Action, index)
-						return &msg, false
+						return &msg, true
 					}
 				}
 			}
 		}
 	}
-	return nil, false
+	return nil, true
```

Comment:
Add a comment to the function about the return values

Reply:
Added a comment